BEIMA.Backend/MongoService/BaseDevice.cs

```diff
@@ -0,0 +1,174 @@
+﻿using System;
+using MongoDB.Bson;
+using MongoDB.Bson.Serialization.Attributes;
+
+namespace BEIMA.Backend.MongoService
+{
+    /// <summary>
+    /// This class represents a Device. This object contains all the required fields necessary
+    /// for a device. This is meant to be used to convert data received from an endpoint, back into a BSON object.
+    /// </summary>
+    public class BaseDevice
+    {
+
+        /// <summary>
+        /// Empty constructor, this allows for a device to be instantiated through Object Initializers.
+        /// </summary>
+        public BaseDevice()
+        {
+            Fields = new BsonDocument();
+            Location = new BsonDocument();
+            LastModified = new BsonDocument();
+        }
+
+        /// <summary>
+        /// Full constructor to instantiate a Device object.
+        /// </summary>
+        /// <param name="id">ObjectId of the object itself.</param>
+        /// <param name="deviceTypeId">ObjectId of the DeviceType this Device is linked to.</param>
+        /// <param name="deviceTag"></param>
+        /// <param name="manufacturer"></param>
+        /// <param name="modelNum"></param>
+        /// <param name="serialNum"></param>
+        /// <param name="yearManufactured"></param>
+        /// <param name="notes"></param>
+        public BaseDevice(ObjectId id, ObjectId deviceTypeId, string deviceTag, string manufacturer, string modelNum, string serialNum, int yearManufactured, string notes)
```

Comment:
```suggestion
        public BaseDevice(ObjectId id, ObjectId deviceTypeId, int yearManufactured, string deviceTag = string.Empty, string manufacturer = string.Empty, string modelNum = string.Empty, string serialNum = string.Empty, string notes = string.Empty)
```
We can set defaults for constructor parameters, also should probably split this into a couple lines.

Reply:
I think you may still be able to manually set the parameters to null when you call it if you specifically specify it.

Reply:
@josephmoore99 @AshlynAdamson If the programmer can still technically pass in null (either unintentionally or intentionally), then setting defaults for constructor parameters is not a good fix. The ??= syntax I've used in this constructor will take care of both defaults and null values.

Reply:
The way I envisioned the full constructor being used, is that the programmer will set variables like so:
`
var id = ...
var deviceTypeId = ...
var deviceTag = ...
(and then all the other properties) 
`
Then, they will pass in all those parameters into the constructor:
`
var device = new BaseDevice(id, deviceTypeId, deviceTag, manufacturer, modelNum, serialNum, yearManufactured, notes);
`

What if in the first block of code, one of those parameters was set to null (because maybe the front end has a bug that made it send a null)? Then, my current implementation will handle this properly and set a default value.

Reply:
I think it's nice to have though because it means you aren't forced to specify each parameter every single time, especially if you don't have the associated values.

Reply:
@josephmoore99 The programmer must specify each parameter each time. There should not be a case where we only specify some fields and not all. The endpoints we write will be generalized, and we must assume that all of the fields will be populated with some sort of data. If it turns out that one field is null, it is okay since the constructor will handle it properly.

Reply:
Alright, if that's the case we can leave it as is. It was more of a quality of life improvement.

Reply:
I definitely understand the point about quality of life, since typing all of those parameters is such a pain. However, in this case, we would have to sacrifice some code/type safety which I am not very comfortable with on such a critical object.